### PR TITLE
Calculator page

### PR DESCRIPTION
**Pull request by marinaprozorovska**

### CODE REVIEW

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+    public class CalculatorPageTests
+    {
+        private IWebDriver driver;
+
+        [SetUp]
+        public void SetUp()
+        {
+            var options = new ChromeOptions { AcceptInsecureCertificates = true };
+            driver = new ChromeDriver(options);
+            driver.Url = "https://localhost:5001/Calculator";
```
Comment (varajan) [inline CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs:21]: Nice trick ;)
Another option was to copy here the positive test-case steps.

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+            //arrange
+            IWebElement incomeOutput = driver.FindElement(By.XPath("//th [text() = 'Income: *']"));
+
+            //assert
+            Assert.AreEqual("Income: *", incomeOutput.Text);
+        }
+
+
+        [Test]
+        public void OrderOfAprilMonthBugTest()
```
Comment (varajan) [inline CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs:64]: This test is covered with the 'OrderOfMonth' one, you should delete it.

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+            IWebElement startMonthField = driver.FindElement(By.Id("month"));
+            
+
+            SelectElement startMonthDropdown = new SelectElement(startMonthField);
+            foreach (IWebElement element in startMonthDropdown.Options)
+            {
+                actualMonths.Add(element.Text);
+            }
+            Assert.AreEqual(expectedMonths, actualMonths);
+           //* Assert.AreEqual(12, startMonthDropdown.Options.Count);
```
Comment (varajan) [inline CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs:91]: No commented code, please.

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+        }
+
+        [Test]
+        public void OrderOfMonth()
+        {
+            List<string> expectedMonths = new List<string> { "January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December" };
+            List<string> actualMonths = new List<string>();
+            IWebElement startMonthField = driver.FindElement(By.Id("month"));
+            
+
```
Comment (varajan) [inline CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs]: Don't leave so much space. Please remove one or both empty lines, please.

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+            new WebDriverWait(driver, TimeSpan.FromSeconds(10))
+            .Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementIsVisible(By.Id("month")));
+            IWebElement aprilMonth = monthDropdown.FindElements(By.TagName("option"))[3];
+
+            //assert
+            Assert.AreEqual("April", aprilMonth.Text);
+        }
+
+        [Test]
+        public void OrderOfMonth()
```
Comment (varajan) [inline CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs]:
```suggestion
        public void OrderOfMonthsTest()
```

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+            {
+                actualMonths.Add(element.Text);
+            }
+            Assert.AreEqual(expectedMonths, actualMonths);
+           //* Assert.AreEqual(12, startMonthDropdown.Options.Count);
+        }
+
+
+        [Test]
+        public void JuneMonthDisplayBugTest()
```
Comment (varajan) [inline CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs:95]: This test is covered with 'OrderOfMonth' test, you should delete this one.

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+            IWebElement rateOfInterestField = driver.FindElement(By.Id("percent"));
+            IWebElement investmentTermField = driver.FindElement(By.Id("term"));
+            IWebElement startDayField = driver.FindElement(By.Id("day"));
+            IWebElement startMonthField = driver.FindElement(By.Id("month"));
+            IWebElement startYearField = driver.FindElement(By.Id("year"));
+            IWebElement financialYearButton365 = driver.FindElement(By.XPath("//tr[@id = 'finYear']/td[text() = '365 days']/*"));
+            IWebElement calculateButton = driver.FindElement(By.Id("calculateBtn"));
+            IWebElement interestEarnedField = driver.FindElement(By.Id("interest"));
+            IWebElement incomeField = driver.FindElement(By.Id("income"));
+            IWebElement endDateField = driver.FindElement(By.Id("endDate"));
```
Comment (varajan) [inline CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs]: Please update the locators to find the inputs via their labels (using xpath) and delete obsolete tests: DepositAmountFieldBugTest, InterestEarnedOutputBugTest, IncomeOutputBugTest

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+
+            //act
+            depositAmountField.SendKeys("100000");
+            rateOfInterestField.SendKeys("100");
+            investmentTermField.SendKeys("365");
+            SelectElement startDayDropdown = new SelectElement(startDayField);
```
Comment (varajan) [inline CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs]: Please move this line to the `arrange` section

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs [hunk 9]
```diff
@@ -0,0 +1,320 @@
...
+            startDayDropdown.SelectByText("1");
+            SelectElement startMonthDropdown = new SelectElement(startMonthField);
```
Comment (varajan) [inline CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs]: Please move this line to the `arrange` section

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs [hunk 10]
```diff
@@ -0,0 +1,320 @@
...
+            startMonthDropdown.SelectByText("January");
+            SelectElement startYearDropdown = new SelectElement(startYearField);
```
Comment (varajan) [inline CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs]: Please move this line to the `arrange` section

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs [hunk 11]
```diff
@@ -0,0 +1,320 @@
...
+            new WebDriverWait(driver, TimeSpan.FromSeconds(20))
+            .Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementToBeClickable(By.Id("calculateBtn")));
+
+            //assert
+            Assert.AreEqual("100,000.00", interestEarnedField.GetAttribute("value"));
+            Assert.AreEqual("200,000.00", incomeField.GetAttribute("value"));
+            Assert.AreEqual("01/01/2023", endDateField.GetAttribute("value"));
+        }
+
+        /* [Test]
```
Comment (varajan) [inline CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs:166]: Commented code.

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs [hunk 12]
```diff
@@ -0,0 +1,320 @@
...
+          }
+
+          string readOnly = startMonthField.GetAttribute("readOnly");
+
+          Assert.AreEqual("true", readOnly);
+
+         */
+
+
+
```
Comment (varajan) [inline CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs]: Too many blank lines. Should be one.

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs [hunk 13]
```diff
@@ -0,0 +1,320 @@
...
+        {
+            //arrange
+            IWebElement depositAmountField = driver.FindElement(By.Id("amount"));
+            IWebElement rateOfInterestField = driver.FindElement(By.Id("percent"));
+            IWebElement investmentTermField = driver.FindElement(By.Id("term"));
+            IWebElement startDayField = driver.FindElement(By.Id("day"));
+            IWebElement startMonthField = driver.FindElement(By.Id("month"));
+            IWebElement startYearField = driver.FindElement(By.Id("year"));
+            IWebElement financialYear = driver.FindElement(By.Id("finYear"));
+            IWebElement financialYearButton365 = financialYear.FindElement(By.XPath("//tr[@id = 'finYear']/td[text() = '365 days']/*"));
```
Comment (varajan) [inline CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs]: You may add a `finYear` parameter to the test-case and use it in XPath. So you can merge this and next test-cases.
Check `Интерполяция строк` https://metanit.com/sharp/tutorial/7.5.php

```suggestion
            IWebElement financialYearButton = financialYear.FindElement(By.XPath($"//tr[@id = 'finYear']/td[text() = '{finYear} days']/*"));
```

**Review by varajan — COMMENTED** [review]
Good work

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs (hunk 1 above)
Comment (marinaprozorovska) [reply CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs:21]: Noted

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs (hunk 3 above)
Comment (marinaprozorovska) [reply CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs:91]: Noted

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs (hunk 5 above)
Comment (marinaprozorovska) [reply CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs]: Fixed

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs (hunk 11 above)
Comment (marinaprozorovska) [reply CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs:166]: Noted

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs (hunk 12 above)
Comment (marinaprozorovska) [reply CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs]: Fixed

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs (hunk 2 above)
Comment (marinaprozorovska) [reply CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs:64]: I decided to save both tests in order to have two solution of one problem)

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs (hunk 9 above)
Comment (marinaprozorovska) [reply CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs]: Fixed

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs (hunk 10 above)
Comment (marinaprozorovska) [reply CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs]: Fixed

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs (hunk 13 above)
Comment (marinaprozorovska) [reply CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs]: Fixed

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs (hunk 6 above)
Comment (marinaprozorovska) [reply CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs:95]: Noted

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs (hunk 8 above)
Comment (marinaprozorovska) [reply CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs]: Fixed

File: CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs (hunk 7 above)
Comment (marinaprozorovska) [reply CalculatorTests/CalculatorTests/Tests/CalculatorPageTests.cs]: Noted

**Review by varajan — APPROVED** [review]
Great

**Pull request merged**